packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts

```diff
@@ -0,0 +1,75 @@
+import { SelectableToken } from '@cardstack/boxel/components/boxel/input/selectable-token';
+import { tracked } from '@glimmer/tracking';
+import { isAddress } from 'web3-utils';
+
+function isNumeric(str: unknown) {
+  if (typeof str !== 'string') return false; // we only process strings!
+  return (
+    !isNaN(str as never) && // use type coercion to parse the _entirety_ of the string (`parseFloat` alone does not do this)...
+    !isNaN(parseFloat(str))
+  ); // ...and ensure strings of whitespace fail
+}
+
+export interface ValidatableForm {
+  selectedPaymentType: 'one-time' | 'monthly' | undefined;
+  paymentDate: Date | undefined;
+  paymentDayOfMonth: number | undefined;
+  monthlyUntil: Date | undefined;
+  recipientAddress: string;
+  paymentAmount: string;
+  paymentToken: SelectableToken | undefined;
+  selectedGasToken: SelectableToken | undefined;
+  maxGasFee: 'normal' | 'high' | 'max' | undefined;
+}
+
+export default class SchedulePaymentFormValidator {
+  form: ValidatableForm;
+
+  @tracked isRecipientAddressInvalid = false;
+  @tracked recipientAddressErrorMessage = '';
+  @tracked isPaymentAmountInvalid = false;
+  @tracked paymentTokenErrorMessage = '';
```

Comment:
Are these used anywhere? Maybe this is a copypaste relic 🤔

Reply:
It is more aspirational :). Open question at the moment: besides disabling the button, what other "invalid" state do we want to show and when?

Reply:
oh yes, having an invalid state without indicating what needs to change can be so frustrating!